contest/templates/contest/scoreboard.html

```diff
@@ -0,0 +1,122 @@
+<div style="float:right">
+  <button type="button" class="btn btn-default" data-toggle="tooltip" data-placement="top" title="Download Scoreboard">
+    <span class="glyphicon glyphicon-download"></span>
+  </button>
+</div>
+<h4>{{ contest.cname }}</h4> {% load staticfiles %}
+<script src="{% static 'contest/js/scoreboard.js' %}"></script>
+<link href="{% static 'contest/css/scoreboard.css' %}" rel="stylesheet">
+{% load contest_extras %}
+<div role="tabpanel">
+  <!-- Nav tabs -->
+  <ul class="nav nav-tabs" role="tablist" id="scoreboardTab">
+    <li role="presentation" class="active"><a href="#testcase" aria-controls="testcase" role="tab" data-toggle="tab">Testcase</a>
+    </li>
+    <li role="presentation"><a href="#penalty" aria-controls="penalty" role="tab" data-toggle="tab">Penalty</a>
+    </li>
+  </ul>
+  <!-- Tab panes -->
+  <div class="tab-content">
+    <!-- testcase -->
+    <div role="tabpanel" class="tab-pane active" id="testcase">
+      <table class="table" style="table-layout:fixed">
+        <thead>
+          <tr>
+            <th>Rank</th>
+            <th>User</th>
+            {% for problem in scoreboard.problems %}
+            <th><b data-toggle="tooltip" title="{{ problem.pname }}">{{ problem.id }}</b>
+            </th>
+            {% endfor %}
+            <th>Total</th>
+          </tr>
+        </thead>
+        <tbody>
+          {% for user in scoreboard|users_sorted_by_solved_testcases %}
+          <tr>
+            <td>{{ forloop.counter }}</td>
+            <td>{{ user.username }}</td>
+            {% for problem in user.problems %}
+            <td>
+              {{ problem|testcases_solved }}/{{ scoreboard|get_problem:problem.id|total_testcase }}
+            </td>
+            {% endfor %}
+            <td>{{ user|testcases_solved }}</td>
+          </tr>
+          {% endfor %}
+          <tr>
+            <td>Total</td>
+            <td></td>
+            {% for problem in scoreboard.problems %}
+              <td>
+                {{ problem.pass_user }}
+              </td>
+            {% endfor %}
+            <td></td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+    <!-- penalty  -->
+    <div role="tabpanel" class="tab-pane" id="penalty">
+      <table class="table" style="table-layout:fixed">
+        <thead>
+          <tr>
+            <th>Rank</th>
+            <th>User</th>
+            {% for problem in scoreboard.problems %}
+            <th><b data-toggle="tooltip" title="{{ problem.pname }}">{{ problem.id }}</b>
+            </th>
+            {% endfor %}
+            <th>Solved</th>
+            <th>Penalty</th>
+          </tr>
+        </thead>
+        <tbody>
+          {% for user in scoreboard|users_sorted_by_penalty %}
+          <tr>
+            <td>{{ forloop.counter }}</td>
+            <td>{{ user.username }}</td>
+            {% for problem in user.problems %}
+            <td>{{ problem|is_solved }}/{{ problem|submit_times }}</td>
```

Comment:
Numerator should be first solved time(in minutes). If a problem is not solved yet, print -- instead.